new_core/filelib.js

```diff
@@ -0,0 +1,94 @@
+/* Json File Library
+ *
+ * A library for handling files with helper function for JSON type files
+ */
+
+const json = {
+	request: undefined,
+	busy: 0,
+	data: [],
+	queue: []
+}
+
+export function addFile(fileurl) {
+
+	// This will add things to the queue if things get busy
+	if(json.busy == 1){
+		json.queue.push(fileurl);
+		return;
+	}
+
+	json.busy = 1
+
+	// This will get the right object for XMLHttpRequest
+	if(window.XMLHttpRequest){
+		alert("Gets here")
+		json.request = new XMLHttpRequest();
+	}else if(window.ActiveXObject){
```

Comment:
we have no IE support, so we should get rid of any ActiveXObject mechanic

Reply:
Yeah, I also agree with this change. Internet Explorer, a relic from an older time.